### DISCUSSION TITLE
How to keep label behavior together with mui and react-hook-form?

### DISCUSSION THREAD

**Discussion by eno-tech0** [uNmAnNeR/imaskjs#920] [Q&A]
I need the ability to show the mask placeholder when focusing on an input. I tried to implement it via onFocus/onBlur, but got unexpected behavior.
If the field does not have a default value, then when a value is entered and unfocused, the label is lowered (expected behavior: the label remains raised if a value is entered in the field).
If the field has a default value and the entire value is erased - the label is not lowered (expected behavior: the label is lowered if the entire value is erased).
How can I get the expected behavior?

Here's an example: https://codesandbox.io/s/imask-mui-react-hook-form-swrc3w?file=/src/App.js

**Comment by uNmAnNeR** [comment]
hi. i don't think it's related to imask functionality. My assumption is that a form does not know when value is changed inside input. But i do not use react (:

**Reply by eno-tech0** [reply]
If you just set lazy to false and don't control it from outside, labels are going up (but placeholder is always displayed, andI need it to be displayed only when the focus is on).
